src/BusinessLayer/TelemetryReporting.cpp

```diff
@@ -241,3 +252,53 @@ void TelemetryReporting::sendData(const unsigned char* data, int length)
       outputDevice_.putChar(data[i]);
    }
 }
+
+void TelemetryReporting::attemptConnection()
+{
+    SerialPortPeripheral SerialPortPeripheral(serialPort_,view_->getCommunicationPort());
+    view_->setConnectionStatus(SerialPortPeripheral.attemptConnection());

```

Comment:
This code seems a bit odd, since the `TelemetryReporting` already has a private member for the serial port, here it would be created again as a local member. Also the name is misleading, variables should be kept in camelCase, not CapCase as it is here.